src/lib/cacheProvider.ts

```diff
@@ -0,0 +1,81 @@
+import { getEnumNameFromValue, StandardEnum } from "./enum";
+import { isNullOrWhitespace } from "./string";
+
+/**
+ * Methods to help caching data of any type.
+ * Data is put in containers and may be identified by a string key.
+ * Data can be saved, retrieved and reset.
+ * Based on Neolution.Abstractions.Caching.CacheProvider<TEnum>
+ * Ref: https://github.com/neolution-ch/Neolution.Abstractions/blob/main/Neolution.Abstractions/Caching/CacheProvider.cs
+ */
+export class CacheProvider<TEnum> {
+  /**
+   * The enum variable
+   */
+  private enumVariable: StandardEnum<TEnum>;

```

Comment:
@drebrez 

I think those variables could be `readonly`

Reply:
Marked as readonly as suggested 👍🏼